lib/gemstash/web.rb

```diff
@@ -18,7 +18,8 @@ def initialize(gemstash_env: nil, http_client_builder: nil)
     end
 
     def http_client_for(server_url)
-      @http_client_builder.for(server_url)
+      timeout = Gemstash::Configuration.new[:fetch_timeout] || 20
+      @http_client_builder.for(server_url, timeout)
```

Comment:
This won't necessarily get the right config, you need to use the config in the environment, so it would be `@gemstash_env.config[:fetch_timeout]` or `Gemstash::Env.current.config[:fetch_timeout]`. Is there a need to vary the timeout from the config? If not, I think it might be nicer if the `Gemstash::HTTPClient` fetched the timeout from the config directly :-)

There should be no need to hard code the default, the config object handles that via `Gemstash::Configuration::DEFAULTS`.